packages/dds/tree/src/core/revertible/revertible.ts

```diff
@@ -22,13 +22,22 @@ export interface Revertible {
 		readonly isLocal: boolean;
 	};
 	/**
-	 * Can be called in order to revert a change. A successful revert will automatically discard resources.
+	 * The current status of the revertible.
 	 */
-	revert(): RevertResult;
+	readonly status: RevertibleStatus;
 	/**
-	 * Should be called to garbage collect any resources associated with the revertible.
+	 * Reverts the associated change and decrements the reference count of the revertible.
 	 */
-	discard(): DiscardResult;
+	revert(): RevertibleResult;
+	/**
+	 * Increments the reference count of the revertible.
+	 * Should be called to prevent/delay the garbage collection of the resources associated with this revertible.
+	 */
+	retain(): RevertibleResult;
```

Comment:
nit: this name feels weird to me but the only alternatives I can think of are "store" or "reference" which aren't great either, could also be very explicit and go with "increaseReferenceCount"

Reply:
Ah, I thought `discard` was a little weird. I'm following the naming in current spec but that spec will need to be re-approved by more people. I've added a comment on the spec to ensure we discuss the names next time we go over it.